hudi-io/src/main/java/org/apache/hudi/io/compress/airlift/HoodieAirliftGzipDecompressor.java

```diff
@@ -50,4 +52,12 @@ public int decompress(InputStream compressedInput,
       return readFully(stream, targetByteArray, offset, length);
     }
   }
+
+  public byte[] compress(byte[] data) throws IOException {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(byteArrayOutputStream)) {
+      gzipOutputStream.write(data);
+    }
+    return byteArrayOutputStream.toByteArray();
```

Comment:
Is there a need to return the compressed bytes?  Is it better to pass in the output stream and write directly to that?

Reply:
Sg.  It looks like we need to know the length of compressed bytes first while writing the HFile data block, so getting the compressed bytes first is unavoidable.